build/cmake/CCache.cmake

```diff
@@ -12,11 +12,29 @@
 find_program (CCACHE_EXECUTABLE ccache)
 if (CCACHE_EXECUTABLE)
     message (STATUS "Found ccache: ${CCACHE_EXECUTABLE}")
-    option (MONGO_USE_CCACHE "Use CCache when compiling" ON)
+
+    execute_process (
+        COMMAND ${CCACHE_EXECUTABLE} --version | perl -ne "print $1 if /^ccache version (.+)$/"
+        OUTPUT_VARIABLE CCACHE_VERSION
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+
+    # Assume `ccache --version` mentions a simple version string, e.g. 1.2.3
+    set (SIMPLE_SEMVER_REGEX "([0-9]+)\.([0-9]+)\.([0-9]+)")
```

Comment:
Good info, thank you. Adjusted the regex to permit omission of the patch number. Also included a CMake warning message to report when version string assumptions are violated so that the need for adjustments can be more easily detected.